File: back/src/main/java/fr/checkpoint/back/entities/Song.java
package fr.checkpoint.back.entities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Song {
	@Id
	@GeneratedValue( strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String name;
	private String artist;
	private String album;
	private String img;

	//	Constructeur
	public Song() {}
	public Song( Long param_id, String param_name, String param_artist, String param_album, String param_img) {
		id = param_id;
		name = param_name;
		artist = param_artist;
		album = param_album;
		img = param_img;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getArtist() {
		return artist;
	}
	public void setArtist(String artist) {
		this.artist = artist;
	}
	public String getAlbum() {
		return album;
	}
	public void setAlbum(String album) {
		this.album = album;
	}
	public String getImg() {
		return img;
	}
	public void setImg(String img) {
		this.img = img;
	}
	public Long getId() {
		return id;
	}
}

File: front/eTunes/src/app/song.ts
export class Song {

    public id:number;
    public name:string;
    public artist:string;
    public album: string;
    public img:string;
    private updateButton:boolean;
    
    constructor (name, artist, album, img) {
        
        this.name = name;
        this.artist = artist;
        this.album = album;
        this.img = img;

        this.updateButton = false;
    }

    getUpdateButton() {
        return this.updateButton
    }
    setUpdateButton( updateButton:boolean) {
        this.updateButton = updateButton;
    }

}
